tests/components/igloohome/conftest.py

```diff
@@ -0,0 +1,73 @@
+"""Common fixtures for the igloohome tests."""
+
+from collections.abc import Generator
+from unittest.mock import AsyncMock, patch
+
+from igloohome_api import GetDeviceInfoResponse, GetDevicesResponse
+import pytest
+
+from homeassistant.components.igloohome.const import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from tests.common import MockConfigEntry
+
+GET_DEVICE_INFO_RESPONSE_LOCK = GetDeviceInfoResponse(
+    id="123456",
+    type="Lock",
+    deviceId="OE1X123cbb11",
+    deviceName="Front Door",
+    pairedAt="2024-11-09T11:19:25+00:00",
+    homeId=[],
+    linkedDevices=[],
+    batteryLevel=100,
+)
+
+
+@pytest.fixture
+def mock_setup_entry() -> Generator[AsyncMock]:
+    """Override async_setup_entry."""
+    with patch(
+        "homeassistant.components.igloohome.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        yield mock_setup_entry
+
+
+@pytest.fixture
+def auth_successful():
+    """Set up the Auth module to always successfully operate."""
+    return patch(
+        "igloohome_api.Auth.async_get_access_token",
+        return_value="mock_access_token",
+    )
+
+
+@pytest.fixture
+def mock_config_entry(hass: HomeAssistant) -> MockConfigEntry:
+    """Return the default mocked config entry."""
+    return MockConfigEntry(
+        title="Client Credentials",
+        domain=DOMAIN,
+        version=1,
+        data={"client_id": "client_id", "client_secret": "client_secret"},
+        unique_id="aa:bb:cc:dd:ee:ff",
+    )
+
+
+@pytest.fixture(autouse=True)
+def api_single_lock():
```

Comment:
missing return type and I'd just call that `mock_api` as you might want to modify the response in later tests, so the name wouldn't be true anymore

Reply:
Hey @zweckj , I'd like to not keep it generic as my intent is to keep the fixture's name descriptive to the "state" the mock is representing. 

I feel that a generic `mock_api` is not descriptive of what can be expected of the mock, so I would like to keep the name and re-use this in the future to re-present more possible states.

Reply:
Since this is the one which is autoused I suggested `mock_api` because that is going to be the default state for all your tests. If you then need to manipulate you can just do `mock_api.property = something` and the name is still true

Reply:
Ah okay, with `autouse` that would make alot of sense. Thanks.